src/pages/ViewPage.tsx

```diff
@@ -62,31 +62,90 @@ export default function ViewPage() {
 	// START - Example on how to apply a custom material to an output
 	/////	
 	const outputNameOrId = "Shelf";
+
+	// create parameter definitions for the custom material
+	const definitions: { [key: string]: IGenericParameterDefinition } = {
+		color: {
+			definition: {
+				id: "colorParam",
+				name: "Custom color",
+				defval: "0x0d44f0ff",
+				type: "Color",
+				hidden: false
+			}
+		},
+		map: {
+			definition: {
+				id: "mapParam",
+				name: "Custom map",
+				defval: "",
+				type: "String",
+				hidden: false
+			}
+		},
+		roughness: {
+			definition: {
+				id: "roughnessParam",
+				name: "Custom roughness",
+				defval: "0",
+				type: PARAMETER_TYPE.FLOAT,
+				min: 0,
+				max: 1,
+				decimalplaces: 4,
+				hidden: false
+			}
+		}
+	};
 	
 	// define a generic parameter which influences a custom material definition
-	const [materialParameters] = useState<IGenericParameterDefinition>({
-		definition: {
-			id: "myparam",
-			name: "Custom color",
-			defval: "0xffffffff",
-			type: "Color",
-			hidden: false
-		}
+	const [materialParameters] = useState<IGenericParameterDefinition[]>(Object.values(definitions));
```

Comment:
@MajorMeerkatThe3rd we can get rid of this state, it seems not to be necessary. I suggest to redefine `definitions` to be an `IGenericParameterDefinition[]`, and to use constants for the parameter names (which we reference in the executor function), and the default values which are used by the default state of `materialProperties`

Reply:
@snabela done in commit https://github.com/shapediver/ShapeDiverReactExample/pull/47/commits/e2fc8efe32be06fffbbbb455c984cc35f7f7ff5c